output/file.go

```diff
@@ -0,0 +1,103 @@
+package output
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/vbauerster/mpb/v5"
+	"github.com/vbauerster/mpb/v5/decor"
+)
+
+type FileWriter struct {
+	FullPath string
+	Filename string
+}
+
+func NewFileWriter(url *url.URL, options *Options) *FileWriter {
+	var tempPath string
```

Comment:
`tempPath`, `tempFile` and `fullPath` can be merged into one variable.

```go
	var fullPath string
	if options.OutputFile == "" {
		fullPath = "./" + filepath.Base(url.Path)
	} else {
		fullPath = options.OutputFile
	}
	if !options.Overwrite {
		fullPath = checkFile(fullPath)
	}
```

Reply:
Done.